src/EditorFeatures/Core/InlineRename/UI/Adornment/RenameFlyoutViewModel.cs

```diff
@@ -229,7 +232,10 @@ public bool Submit()
         }
 
         SmartRenameViewModel?.Commit(IdentifierText);
-        Session.InitiateCommit();
+
+        var token = _asyncListener.BeginAsyncOperation(nameof(Submit));
+        Session.CommitAsync(previewChanges: false, editorOperationContext: null)
+            .ReportNonFatalErrorAsync().CompletesAsyncOperation(token);
```

Comment:
nit: consider mentioning that CommitAsync will display a proper UI to the user while this async work is happening.